waltz-server/src/main/java/com/wepay/waltz/store/internal/StoreSessionImpl.java

```diff
@@ -165,6 +165,15 @@ public void append(StoreAppendRequest request) throws SessionClosedException, Ge
                     throw new GenerationMismatchException();
                 }
 
+                // Wait until the queue has enough space.
+                while (requestQueue.size() > BATCH_SIZE * 2) {
```

Comment:
Need to break out of the loop when `running` became false.

Reply:
Don!

Reply:
Done!